server_push_preload plugin is not compilable on macOS

I got these compile errors on macOS.

```
experimental/server_push_preload/server_push_preload.cc:129:55: error: no member named 'ptr_fun' in namespace 'std'
      s.erase(find_if(s.rbegin(), s.rend(), not1(std::ptr_fun<int, int>(isspace))).base(), s.end()); // trim left
                                                 ~~~~~^
experimental/server_push_preload/server_push_preload.cc:129:66: error: expected '(' for function-style cast or type construction
      s.erase(find_if(s.rbegin(), s.rend(), not1(std::ptr_fun<int, int>(isspace))).base(), s.end()); // trim left
                                                              ~~~^
experimental/server_push_preload/server_push_preload.cc:129:71: error: expected '(' for function-style cast or type construction
      s.erase(find_if(s.rbegin(), s.rend(), not1(std::ptr_fun<int, int>(isspace))).base(), s.end()); // trim left
                                                                   ~~~^
experimental/server_push_preload/server_push_preload.cc:130:64: error: no member named 'ptr_fun' in namespace 'std'
      s.erase(s.begin(), find_if(s.begin(), s.end(), not1(std::ptr_fun<int, int>(isspace))));        // trim right
                                                          ~~~~~^
experimental/server_push_preload/server_push_preload.cc:130:75: error: expected '(' for function-style cast or type construction
      s.erase(s.begin(), find_if(s.begin(), s.end(), not1(std::ptr_fun<int, int>(isspace))));        // trim right
                                                                       ~~~^
experimental/server_push_preload/server_push_preload.cc:130:80: error: expected '(' for function-style cast or type construction
      s.erase(s.begin(), find_if(s.begin(), s.end(), not1(std::ptr_fun<int, int>(isspace))));        // trim right
                                                                            ~~~^
6 errors generated.

```

The code hasn't been changed since last May. I *guess* it's because of recent toolchain update.

--- a/plugins/experimental/server_push_preload/server_push_preload.cc
+++ b/plugins/experimental/server_push_preload/server_push_preload.cc
@@ -126,8 +126,8 @@ public:
     string s;
     set<string> tokens;
     while (getline(ss, s, delim)) {
-      s.erase(find_if(s.rbegin(), s.rend(), not1(std::ptr_fun<int, int>(isspace))).base(), s.end()); // trim left
-      s.erase(s.begin(), find_if(s.begin(), s.end(), not1(std::ptr_fun<int, int>(isspace))));        // trim right
+      s.erase(find_if(s.rbegin(), s.rend(), [](unsigned char c) { return !std::isspace(c); }).base(), s.end()); // trim left
+      s.erase(s.begin(), find_if(s.begin(), s.end(), [](unsigned char c) { return !std::isspace(c); }));        // trim right
       tokens.insert(s);
     }
     return tokens;

